modules/search/class.jetpack-search.php

```diff
@@ -1125,10 +1207,50 @@ public function convert_wp_es_to_es_args( array $args ) {
 		$es_query_args['filter']       = $parser->build_filter();
 		$es_query_args['query']        = $parser->build_query();
 		$es_query_args['aggregations'] = $parser->build_aggregation();
+		$es_query_args['suggest']      = $this->build_experimental_suggest( $args['query'], $parser );
 
 		return $es_query_args;
 	}
 
+	/**
+	 * Assemble the 'suggest' portion of an ES query, using a phrase suggester. Note that this is experimental and
+	 * only works with English tokens.
+	 *
+	 * @param string $query                    The query text being searched.
+	 * @param Jetpack_WPES_Search_Query_Parser The search query parser for this query.
+	 *
+	 * @return array An experimental suggest query.
+	 */
+	private function build_experimental_suggest( $query, &$parser ) {
+		if ( defined( 'JETPACK_SEARCH_VIP_INDEX' ) && JETPACK_SEARCH_VIP_INDEX ) {
+			// VIP indices do not have per language fields.
+			$field = 'mlt_content';
+		} else {
+			// Add the language suffix but not a boost.
+			$field_with_boost = $parser->merge_ml_fields( array( 'mlt_content' => 0 ), array() )[0];
+			$field = rtrim( $field_with_boost, '^0' );
+		}
+		return array(
+			'text' => $query,
+			'experimental-phrase-suggester' => array(
+				'phrase' => array(
+					'field'                      => $field,
+					"max_errors"                 => 4.0,
+					'confidence'                 => .9,
+					'real_word_error_likelihood' => .9,
+					'size' => 1,
+					"direct_generator" => array(
```

Comment:
I think we should add in a collate query here so that we only consider suggestions that actually have a result. I think the query only needs to run the search against the mlt_content field, and then filter by the site's blog_id.

Doing this should speed up the case where:
1) no results for the original query
2) no results for the suggested query

It may also let the suggest return the second or third result when there are no results for the first one. Not sure about that but it would be a good benefit.